nexus_client_sdk/nexus/input/payload_reader.py

```diff
@@ -52,9 +55,36 @@ async def __aenter__(self):
             self._http = session_with_retries()
         http_response = self._http.get(url=self._payload_uri)
         http_response.raise_for_status()
-        self._payload = self._payload_type.from_json(http_response.content)
+
+        payload_dict: dict = DictJsonSerializationFormat().deserialize(http_response.content)
```

Comment:
This adds a potential failure point, as you interpret the payload with 2 different deserializers. Needs change